### ISSUE TITLE
How to add Extra rows before the real data

### ISSUE THREAD

**Issue by gheesixz** [rap2hpoutre/fast-excel#292]
Hi is there any way to add atleast blank rows before the data??
Some website requires to have their custom headers/template present.
In my case, i need to make a custom rows atleast 4 rows before the real data comes in.

**Comment by StefanoTesla** [comment]
Yes really easy, you need to push a new attribute to the array/collection


foreach $datapage as $data{ array_push($data['mycolun']); }

**Comment by elminson** [comment]
There's no template feature, but since the export just writes your collection row by row, prepend whatever rows you want and turn off the auto-generated header:

```php
$custom = collect([
    ['My Company', '', '', ''],
    ['Report title', '', '', ''],
    ['', '', '', ''],
    ['id', 'name', 'email', 'created_at'], // your own header row
]);

$rows = $custom->merge($data->map(fn ($u) => [
    $u->id, $u->name, $u->email, $u->created_at,
]));

(new FastExcel($rows))->withoutHeaders()->export('report.xlsx');
```

Use `withoutHeaders()` so the keys aren't auto-emitted as a header, and put your blank/custom rows first. Closing as answered.
